root/usr/share/fdi/facts/openlldp.rb

```diff
@@ -0,0 +1,116 @@
+# grabbed from:
+# https://raw.githubusercontent.com/razorsedge/puppet-openlldp/master/lib/facter/openlldp.rb
+
+#Copyright (C) 2012 Mike Arnold <mike@razorsedge.org>
+#
+#Licensed under the Apache License, Version 2.0 (the "License");
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+
+# Fact:
+#   lldp_neighbor_chassisid_<interface>
+#   lldp_neighbor_mngaddr_ipv4_<interface>
+#   lldp_neighbor_mngaddr_ipv6_<interface>
+#   lldp_neighbor_mtu_<interface>
+#   lldp_neighbor_portid_<interface>
+#   lldp_neighbor_sysname_<interface>
+#   lldp_neighbor_pvid_<interface>
+#
+# Purpose:
+#   Return information about the host's LLDP neighbors.
+#
+# Resolution:
+#   On hosts with the lldptool binary, send queries to the lldpad for each of
+#   the host's Ethernet interfaces and parse the output.
+#
+# Caveats:
+#   Assumes that the connected Ethernet switch is sending LLDPDUs, Open-LLDP
+#   (lldpad) is running, and lldpad is configured to receive LLDPDUs on each
+#   Ethernet interface.
+#
+# Authors:
+#   Mike Arnold <mike@razorsedge.org>
+#
+# Copyright:
+#   Copyright (C) 2012 Mike Arnold, unless otherwise noted.
+#
+
+require 'facter/util/macaddress'
+
+# http://www.ruby-forum.com/topic/3418285#1040695
+module Enumerable
+  def grep_v(cond)
+    select {|x| not cond === x}
+  end
+end
+
+#if Facter::Util::Resolution.which('lldptool')
+if File.exists?('/usr/sbin/lldptool')
+  lldp = {
+    # LLDP Name    Numeric value
+    'chassisID'    => '1',
+    'portID'       => '2',
+    'sysName'      => '5',
+    'mngAddr_ipv4' => '8',
+    'mngAddr_ipv6' => '8',
+    'PVID'         => '0x0080c201',
+    'MTU'          => '0x00120f04',
+  }
+
+  # Remove interfaces that pollute the list (like lo and bond0).
+  Facter.value('interfaces').split(/,/).grep_v(/^lo$|^bond[0-9]/).each do |interface|
+    # Loop through the list of LLDP TLVs that we want to present as facts.
+    lldp.each_pair do |key, value|
+      Facter.add("lldp_neighbor_#{key}_#{interface}") do
+        setcode do
+          result = ""
```

Comment:
The result must be `nil` otherwise `facter` returns empty facts and we don't want that.

Other that that, it works. But the loop is useless and it also returns empty facts for interfaces which do not support LLDP.

Please fix the two guys here, rebase and we are finally good to go :-)

I also found a UI bug, LLDP facts were not displayed properly. Filed a bug: http://projects.theforeman.org/issues/14010